src/swoop/api/routers/payloads.py

```diff
@@ -42,13 +112,50 @@ def list_payloads(
         "500": {"model": APIException},
     },
 )
-def get_job_status(
-    payload_id: str = Path(..., alias="payloadId"),
+async def get_payload_status(
+    request: Request, payload_id
 ) -> PayloadInfo | APIException:
     """
     retrieve info for a payload
     """
-    pass
+
+    async with request.app.state.readpool.acquire() as conn:
+        q, p = render(
+            """
+                    SELECT
+                    *
+                    FROM swoop.payload_cache c
+                    INNER JOIN swoop.item_payload i
+                    ON c.payload_uuid = i.payload_uuid
+                    INNER JOIN swoop.action a
+                    ON c.payload_uuid = a.payload_uuid
+                    WHERE c.payload_uuid = :payload_id::uuid;
+                """,
+            payload_id=payload_id,
+        )
+        records = await conn.fetch(q, *p)
+
+        if not records:
+            raise HTTPException(
+                status_code=404, detail="No payload that matches payload uuid found"
+            )
+
+        return PayloadInfo(
+            payload_uuid=payload_id,
+            payload_hash=str(records[0]["payload_hash"]),
+            workflow_version=records[0]["workflow_version"],
+            workflow_name=records[0]["workflow_name"],
+            created_at=records[0]["created_at"],
+            invalid_after=records[0]["invalid_after"],
+            collections=list({record["collection"] for record in records}),
+            items=list({record["item_id"] for record in records}),
```

Comment:
Collections and item IDs on their own are mostly meaningless, as how can you know which collection maps to which item? I think this should instead be a single array that might look like this:

```
"items": [
  {
    "item_id": <id>,
    "collection": <collection>
  }
]
```